spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfiguration.java

```diff
@@ -86,4 +97,109 @@ public PrometheusScrapeEndpoint prometheusEndpoint(
 
 	}
 
+	/**
+	 * Configuration for <a href="https://github.com/prometheus/pushgateway">Prometheus
+	 * Pushgateway</a>.
+	 *
+	 * @author David J. M. Karlsen
+	 */
+	@Configuration
+	@ConditionalOnClass(PushGateway.class)
+	@ConditionalOnProperty(prefix = "management.metrics.export.prometheus.pushgateway", name = "enabled", havingValue = "true", matchIfMissing = false)
+	public static class PrometheusPushGatewayConfiguration {
+
+		@Bean
+		public PushGatewayHandler pushGatewayHandler(CollectorRegistry collectorRegistry,
+				PrometheusProperties prometheusProperties, Environment environment) {
+			return new PushGatewayHandler(collectorRegistry, prometheusProperties,
+					environment);
+		}
+
+		public static class PushGatewayHandler {
```

Comment:
I think this should be a private or package private class. It seems like we wouldn't expect people to use it directly.